django-rgd/rgd/rest/viewsets.py

```diff
@@ -49,3 +52,13 @@ def footprint(self, request, pk=None):
 class SpatialAssetViewSet(ModelViewSet):
     serializer_class = serializers.SpatialAssetSerializer
     queryset = models.SpatialAsset.objects.all()
+
+
+class SignatureView(BaseRestViewMixin, views.APIView):
+    """Generate an expirey URL signature."""
+
+    def post(self, request):
+        signer = UserSigner()
+        signature = signer.sign(user=self.request.user)
+        param = getattr(settings, 'RGD_SIGNED_URL_QUERY_PARAM', 'signature')
+        return response.Response({param: signature})
```

Comment:
I think it'd be good to include an `expires_at` param here, indicating the unix timestamp this signature expires as. I'll open an issue so it can be addressed separately.

Reply:
https://github.com/ResonantGeoData/ResonantGeoData/issues/606